kiwi/schema/kiwi.rnc

```diff
@@ -2115,36 +2131,36 @@ div {
         attribute tag { text }
         >> sch:pattern [ id = "tag" is-a = "container_type"
             sch:param [ name = "attr" value = "tag" ]
-            sch:param [ name = "types" value = "docker oci" ]
+            sch:param [ name = "types" value = "docker oci appx" ]
```

Comment:
tags are being used in WSL?

Reply:
It's not and I had the same idea when I implemented it. But I could not come across a way to specify this in the schema. If you take a look at the schema for <containerconfig> you see:

```xml
sch:pattern [
        abstract = "true"
        id = "container_type"
        sch:rule [
            context = "containerconfig"
            sch:assert [
                test = "contains('$types', ../@image)"
                "containerconfig($attr) is only available for the following "~
                "image types: $types"
            ]
        ]
    ]
```

This schematron rule is applied on any attribute of the containerconfig element even if one ore more attributes are NOT specified in the XML description

Now if I go and say

```xml
k.containerconfig.tag.attribute =
        ## Specifies a tag for the container. This is usually the
        ## the tag name of the container as read if the container
        ## image is imported via the docker load command
        attribute tag { text }
        >> sch:pattern [ id = "tag" is-a = "container_type"
            sch:param [ name = "attr" value = "tag" ]
            sch:param [ name = "types" value = "docker oci" ]
        ]
```

and leave out appx I will get a validation error that says:

```
 --> containerconfig(tag) is only available for the following image types: docker oci
[ ERROR   ]: 11:14:11 | KiwiDescriptionInvalid: Schema validation for build-tests/x86/suse/test-image-wsl/appliance.kiwi failed
```

and that happens even if you don't have set a tag in the XML description. So the schema validation in schematron is always applied on all possible elements and attributes. Given this I have no idea how I could specify what you are after

If you know a way I'm happy to add it

Reply:
I believe I got it, there is a problem on the context definition of the schematron rules. Compare it the schematron rule for type element attributes constraints. 

On my tests it works fine with:

```
    sch:pattern [
        abstract = "true"
        id = "container_type"
        sch:rule [
            context = "containerconfig[@$attr]"
            sch:assert [
                test = "contains('$types', ../@image)"
                "containerconfig($attr) is only available for the following "~
                "image types: $types"
            ]
        ]
    ]
```
see the context definition.

With this and removing the `appx` image type for `tags`, `additionaltags`, `maintainer` and `workingdir` I can validate the WSL text XML you provided.

Reply:
My bad I entered this containerconfig back in a time and since the two supported image types were actually using the same config it did not pop up.

Reply:
Aaargh I completely overlooked the context. Yes that works. Thanks much